x/incentive/keeper/program.go

```diff
@@ -0,0 +1,50 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v4/x/incentive"
+)
+
+// CreateIncentiveProgram saves an incentive program to upcoming programs after it
+// passes governance, and also attempts to fund it from the module's community fund
+// address if sufficient funds are available. The program is always added to upcoming
+// even if funding fails or its start date has already passed, but an error is returned
+// instead if it fails validation.
+func (k Keeper) CreateIncentiveProgram(
+	ctx sdk.Context,
+	program incentive.IncentiveProgram,
+	fromCommunityFund bool,
+) error {
+	if err := program.Validate(); err != nil {
+		return err
+	}
+
+	addr := k.GetCommunityFundAddress(ctx)
+	if fromCommunityFund && !addr.Empty() {
```

Comment:
The module should be able to function without a community fund address set - it just defaults to manual funding in that case.

Reply:
In that case we should return error. If a proposal requests funding from community fund, but community fund is not available, then this is an error

Reply:
Do errors in gov message execution cause their effects to be rolled back, or is it a chain halt? If it's a rollback we can make an error for it

Reply:
Right now, it's designed to fail safely - also in practice, we want to be able to recover programs whose funding fails, by funding them manually. This is superior to starting another 5 days governance process for the same proposal

Reply:
> Do errors in gov message execution cause their effects to be rolled back

No, when proposal is executed and causes an error, the proposal is skipped and error is logged.

Reply:
> This is superior to starting another 5 days governance process for the same proposal

I don't think muting an error is a right solution. I would consider following cases, when we request proposal with funding from a community fund: 
* community fund is not available: proposal should fail when submitting (rather than waiting 5 days to fail during the execution)
* community fund doesn't have enough balance: proposal should fail, error should be reported (ideally with appropriate event).

Reply:
I'll add the error reporting right away - as for how the prop should fail, we have two options:

- Program is created but not funded. If the program is not funded manually by its start date, it is cancelled. (Current approach)
- Program fails to even be created. 5 day gov process must be restarted even if all the parameters are the same.

I see no advantage and a large operational disadvantage in switching from the current approach to the second.

Reply:
Added the error checking - currently _non-critical_ failures - that is, when the code detects in advance that community funding will fail and reverting manual funding fixes the problem - are logged as errors but do not cause the proposal itself to fail.